mitodl/mit-open
# News & Events API

### What are the relevant tickets?
Related to #611, based on PR #612 which should be merged first.

### Description (What does it do?)
- Adds API endpoints for news_events sources and items, filterable by feed_type ("news", "events")
- Moves some classes/functions from `learning_resources/filters.py` to `main/filters.py` because they might be useful for multiple API endpoints (and one is used for this new endpoint).


### How can this be tested?
- Run `./manage.py backpopulate_news_events` if you haven't already
- Try the following urls:
  - http://localhost:8063/api/v0/news_events_sources/ - should return 2 results
  -  http://localhost:8063/api/v0/news_events_sources/?feed_type=news,events - should return 2 results
  - http://localhost:8063/api/v0/news_events_sources/?feed_type=news - should return 1 result, with `{"feed_type": "news"}`
  - http://localhost:8063/api/v0/news_events_sources/?feed_type=events - should return 1 result, with `{"feed_type": "events"}`
  - http://localhost:8063/api/v0/news_events/ - should return multiple results for news and events.  Events should come before news, with all items sorted by `event_details.event_datetime` / `news_details.publish_date` in reverse order.
  - http://localhost:8063/api/v0/news_events/?feed_type=news - should return only news items, sorted by `news_details.publish_date` in reverse order
  - http://localhost:8063/api/v0/news_events/?feed_type=events - should return only event items, sorted by `event_details.event_datetime` in reverse order.
